manager/src/main/java/com/redhat/service/bridge/manager/workers/WorkManagerImpl.java

```diff
@@ -0,0 +1,104 @@
+package com.redhat.service.bridge.manager.workers;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.bridge.manager.dao.WorkDAO;
+import com.redhat.service.bridge.manager.models.ManagedResource;
+import com.redhat.service.bridge.manager.workers.id.WorkerId;
+
+import io.quarkus.scheduler.Scheduled;
+import io.vertx.mutiny.core.eventbus.EventBus;
+
+public class WorkManagerImpl implements WorkManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WorkManagerImpl.class);
+
+    @Inject
+    WorkDAO workDAO;
+
+    @Inject
+    EventBus eventBus;
+
+    @Inject
+    WorkerId workerId;
+
+    @Override
+    public Work schedule(ManagedResource managedResource) {
+        Work w = workDAO.findByManagedResource(managedResource);
+        if (w == null) {
+            w = Work.forResource(managedResource, this.workerId.value());
+            if (LOGGER.isInfoEnabled()) {
+                LOGGER.info(String.format("Scheduling work for '%s' [%s]",
+                        w.getManagedResourceId(),
+                        w.getType()));
+            }
+            persist(w);
+            fireEvent(w);
+        } else {
+            w.setScheduledAt(ZonedDateTime.now());
+            persist(w);
+        }
+
+        return w;
+    }
+
+    @Transactional
+    protected void persist(Work work) {
+        workDAO.persist(workDAO.getEntityManager().merge(work));
+    }
+
+    private void fireEvent(Work w) {
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.info(String.format("Executing work for '%s' [%s]",
+                    w.getManagedResourceId(),
+                    w.getType()));
+        }
+        eventBus.requestAndForget(w.getType(), w);
+    }
+
+    @Override
+    @Transactional
+    public boolean exists(Work work) {
+        return Objects.nonNull(workDAO.findById(work.getId()));
+    }
+
+    @Override
+    @Transactional
+    public void complete(Work work) {
+        if (!exists(work)) {
+            return;
+        }
+
+        workDAO.deleteById(work.getId());
+    }
+
+    @SuppressWarnings("unused")
+    @Scheduled(every = "5s", concurrentExecution = Scheduled.ConcurrentExecution.SKIP)
```

Comment:
Well, at the moment it is also the re-try frequency for polling asynchronous activities.

@lucamolteni You was most vocal about how this could affect our SLO. @robpblake @lucamolteni WDYT?

Reply:
Anything is fine by me :)

Reply:
@manstis The SLO can be whatever we want it to be, so I'm really not worried about that right now. As I said before, I'd prefer slow but reliable here in the first instance. 

`5s` feels a bit tight as if anything is even remotely slow in say Managed Kafka or Managed Connectors, we'll hit our retry limit pretty quickly. I'd say `15` or maybe even `30` seconds is a good starting point for us.

Reply:
OK, changed to 30s for production\qa\ITs. 1s for `\manager` tests.